src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,304 @@
+
+import { FormattedMessage } from 'react-intl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+const intl = useIntl();
+
+function recomposeAllRelationshipObjectFromModifiedCategoryObject(
+  currentAllRelationshipObject,
+  modifiedTargetCategoryObj,
+) {
+  const nonTargeCategoryObjects = filter(
+    currentAllRelationshipObject,
+    currentRelationshipCategory =>
+      currentRelationshipCategory.category !==
+      modifiedTargetCategoryObj?.category,
+  );
+  return [...nonTargeCategoryObjects, modifiedTargetCategoryObj];
+}
+
+function deleteCategory(allRelationshipsObj, category) {
+  return allRelationshipsObj.filter(
+    singleRelationshipObj =>
+      singleRelationshipObj?.category !== category,
+  );
+}
+
+function deleteRelationship(
+  allRelationshipsObj,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategoryObjArr = filter(
+    allRelationshipsObj,
+    currentRelationshipCategory =>
+      currentRelationshipCategory.category === currentCategory,
+  );
+  // if (targetRelationshipCategoryObjArr?.length > 1) return; // TODO how do we handle duplicate relationship categories/types?
+  const targetRelationshipCategoryObj = get(
+    targetRelationshipCategoryObjArr,
+    '0',
+    {},
+  );
+  const modifiedArr = targetRelationshipCategoryObj.relationships.filter(
+    entry => entry !== relationship,
+  );
+  const modifiedTargetCategoryObj = {
+    category: currentCategory,
+    relationships: modifiedArr,
+  };
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    modifiedTargetCategoryObj,
+  );
+}
+
+function updateRelationshipInCategory(
+  allRelationshipsObj,
+  category,
+  oldRelationshipName,
+  newRelationshipName,
+) {
+  const targetRelationshipCategoryObj = get(
+    filter(
+      allRelationshipsObj,
+      currentRelationshipCategory =>
+        currentRelationshipCategory?.category === category,
+    ),
+    '0',
+    {},
+  );
+  const currentRelationships =
+    targetRelationshipCategoryObj?.relationships;
+  const oldRelationshipIndex = currentRelationships?.indexOf(
+    oldRelationshipName,
+  );
+  if (oldRelationshipIndex > -1)
+    currentRelationships[oldRelationshipIndex] = newRelationshipName;
+  targetRelationshipCategoryObj.relationships = currentRelationships;
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    targetRelationshipCategoryObj,
+  );
+}
+
+function createCategory(allRelationshipsObj) {
+  const newCategoryObj = {
+    category: intl.formatMessage({ id: 'ENTER_NEW_CATEGORY_NAME' }),
```

Comment:
I would prefer to not have to delete this text. When adding a region, we just add an empty text field. If you want to make it clear that this level of text field is a category and not a relationship, I think that a placeholder would be better than setting the actual value.

Reply:
Done